### PR TITLE
Fix compatibility issues with several timeseries primitives

### LINKED ISSUE #89: Negative index anomalies in timeseries_errors
`mlprimitives/candidates/timeseries_errors.py`: there is a problem in which the predicted anomalies have a negative index.

When printing the anomalies, it seems that we are getting the following intervals:

[-453, 100]
[151, 350]
[400, 468]

The first interval is incorrect, and is a result of not doing a check in the index that is being checked for anomalies.

### LINKED ISSUE #90: JSON of LSTMTimeSeriesRegressor doesn't match description
Currently the json file `mlprimitives/jsons/keras.Sequential.LSTMTimeseriesRegressor.json` doesn't match the description of being a single layer LSTM network.

A change needs to be made so the model has one LSTM, one dropout, and one dense output layers, since the use of dropout in the LSTM layer was deprecated and generates a warning if not addressed.

### LINKED ISSUE #91: Issue with timeseries.py json files
Currently, when trying to use the `rolling_window_average.json` and `time_segments_average.json`, there's an issue because the context variables are not set. 

For `mlprimitives/jsons/mlprimitives.candidates.timeseries.rolling_window_average.json` and `/time_segments_average.json` parameters for `time_column` and `value_column` must be added so the MLBlocks context works.

### PR DESCRIPTION
**Pull request by itinawi**
Resolves #89
Resolves #90
Resolves #91

### CODE REVIEW

File: mlprimitives/jsons/keras.Sequential.LSTMTimeSeriesRegressor.json [hunk 1]
```diff
@@ -67,16 +76,19 @@
...
                 "type": "list",
                 "default": [
                     {
-                        "class": "keras.layers.Input",
+                        "class": "keras.layers.LSTM",
                         "parameters": {
-                            "shape": "input_shape"
+                            "input_dim": "input_dim",
+                            "input_length": "input_length",
+                            "units": "lstm_hidden_units",
```
Comment (csala) [inline mlprimitives/jsons/keras.Sequential.LSTMTimeSeriesRegressor.json]: Please call this `lstm_units`, following the pattern `{layer_name}_{parameter_name}`

File: mlprimitives/jsons/mlprimitives.candidates.timeseries.rolling_window_sequences.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "mlprimitives.candidates.timeseries.rolling_window_sequences",
-    "contributors": ["Ihssan Tinawi <itinawi@mit.edu>"],
+    "author": "Ihssan Tinawi <itinawi@mit.edu>",
```
Comment (csala) [inline mlprimitives/jsons/mlprimitives.candidates.timeseries.rolling_window_sequences.json]: Please don't replace the `contributors` entry.

**Comment by csala** [comment]
@itinawi The changes in the code look good, but I want to wait for the issues to be reported properly so we can keep track of what this PR is actually resolving, so I keep it unmerged for now

File: mlprimitives/candidates/timeseries.py [hunk 3]
```diff
@@ -27,8 +27,8 @@ def rolling_window_sequences(X, window_size, value_column, time_column):
     time = []
     for i in range(len(X) - window_size):
         # reshape into a vector to fit into a neural network model (vectorize it)
-        output_X.append(X[i: i + window_size][value_column].values.reshape([-1, 1]))
-        y.append(X[i + window_size + 1][value_column].values.reshape([-1, 1]))
+        output_X.append(X[i: i + window_size][value_column].values.copy().reshape([-1, 1]))
```
Comment (csala) [inline mlprimitives/candidates/timeseries.py]: Why have been these `copy` statements added here?
This doesn't seem to be part of any of the issues that this PR is trying to solve.
Please review if this can be removed so that there are no changes to this file.
